src/AuthContext.jsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { createContext, useState, useEffect } from 'react';
+import { onAuthStateChanged } from 'firebase/auth';
+import { auth } from './api/config';
+
+export const AuthContext = createContext();
+
+export default function AuthProvider({ children }) {
+	const [currentUser, setCurrentUser] = useState(null);
+	const [pending, setPending] = useState(true);
+
+	useEffect(() => {
+		onAuthStateChanged(auth, (user) => {
+			setCurrentUser(user);
+			setPending(false);
+		});
+	}, []);
+
```

Comment:
nice use of React Context 💫. We already have this code, for fetching and detecting user changes, in the `useAuth` custom hook so this is a duplication. I would recommend to move the pending state/logic to the `useAuth` hook and use the hook here.

Reply:
Thank you Reda for pointing that out. I took a closer look at the code and i figured that the using a context wasn't necessary so I removed the context as a whole. The route protection works just fine regardless. Please review